crates/core/component/stake/src/identity_key.rs

```diff
@@ -18,11 +21,83 @@ use decaf377_rdsa::{SpendAuth, VerificationKey};
 ///
 /// Using a [`SpendAuth`] key means that validators can reuse code and processes
 /// designed for custodying funds to protect their identity.
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(try_from = "pb::IdentityKey", into = "pb::IdentityKey")]
-pub struct IdentityKey(pub VerificationKey<SpendAuth>);
+pub struct IdentityKey {
+    /// The *compressed* bytes of the identity verification key.
+    bytes: VerificationKeyBytes<SpendAuth>,
+    // The *decompressed* identity verification key.
+    key: OnceLock<DecompressionResult>,
+}
+
+type DecompressionResult = Result<VerificationKey<SpendAuth>, decaf377_rdsa::Error>;
+
+impl PartialOrd for IdentityKey {
+    /// [`IdentityKey`]s are partially ordered according to their compressed bytes.
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.bytes.partial_cmp(&other.bytes)
+    }
+}
+
+impl Ord for IdentityKey {
+    /// [`IdentityKey`]s are totally ordered according to their compressed bytes.
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.bytes.cmp(&other.bytes)
+    }
+}
+
+impl From<SigningKey<SpendAuth>> for IdentityKey {
+    /// An [`IdentityKey`] can be constructed from a [`decaf377_rdsa::SigningKey`].
+    fn from(sk: SigningKey<SpendAuth>) -> Self {
+        let vk: VerificationKey<SpendAuth> = sk.into();
+        Self::from(vk)
+    }
+}
+
+impl<'sk> From<&'sk SigningKey<SpendAuth>> for IdentityKey {
+    /// An [`IdentityKey`] can be constructed from a reference to a [`decaf377_rdsa::SigningKey`].
+    fn from(sk: &'sk SigningKey<SpendAuth>) -> Self {
+        Self::from(*sk)
+    }
+}
+
+impl From<VerificationKey<SpendAuth>> for IdentityKey {
+    /// An [`IdentityKey`] can be constructed from a [`decaf377_rdsa::VerificationKey`].
+    fn from(vk: VerificationKey<SpendAuth>) -> Self {
+        let bytes: VerificationKeyBytes<SpendAuth> = vk.into();
+        let key: OnceLock<_> = Ok(vk).into();
+
+        Self { bytes, key }
+    }
+}
+
+impl<'vk> From<&'vk VerificationKey<SpendAuth>> for IdentityKey {
+    /// An [`IdentityKey`] can be constructed from a reference to a [`decaf377_rdsa::VerificationKey`].
+    fn from(vk: &'vk VerificationKey<SpendAuth>) -> Self {
+        Self::from(*vk)
+    }
+}
```

Comment:
I'm not sure it's desirable to have all of these impls. We didn't have them before, do we need them to change the internal representation?